src/controllers/warehousecontroller.js

```diff
@@ -54,6 +56,9 @@ class WarehouseController extends Controller
                     }
                 }
             );
+
+            var history = new HistoryController();
```

Comment:
Please use descriptive names, like `historyController`. That's more consistent with the other variable names.